### PR TITLE
[Config] Notify unknown field in genesis.json

### PR DESCRIPTION
**Pull request by hyunsooda**
## Proposed changes

Earlier, certain unspecified fields in genesis.json were overlooked without issuing a clear error message. Given the increasing size of genesis.json, displaying typos might be convenient. If this sanitization is undesired, please let me know it. 

Given a `genesis.config`,
```
{
    "config": {
        ...
        "randaoCompatibleBlock123": 1242,
       ...
```

Outputs error message without program exit.
```
ERROR[11/16,10:58:38 +09] [19] Unmarshal error                           error="json: unknown field \"randaoCompatibleBlock123\""
WARN[11/16,10:58:38 +09] [30] Override the default governance config to the chain config
INFO[11/16,10:58:38 +09] [18] Setting connection type                   nodetype=cn conntype=0
INFO[11/16,10:58:38 +09] [18] Cypress bootnodes are set
...
```

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [ ] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: blockchain/gen_genesis.go [hunk 1]
```diff
@@ -60,6 +62,12 @@ func (g *Genesis) UnmarshalJSON(input []byte) error {
...
 	var dec Genesis
+
+	decoder := json.NewDecoder(bytes.NewReader(input))
+	decoder.DisallowUnknownFields()
+	if err := decoder.Decode(&dec); err != nil {
+		log.NewModuleLogger(log.CMDUtilsNodeCMD).Error("Unmarshal error", "error", err)
+	}
 	if err := json.Unmarshal(input, &dec); err != nil {
 		return err
 	}
```
Comment (hyeonLewis) [inline blockchain/gen_genesis.go:73]: Do we still need this? `dec` is already unmarshal in `decoder.Decode(&dec)`.

Comment (hyunsooda) [reply blockchain/gen_genesis.go:73]: Only the "unknown field" is acceptable error in `Decode()`, but there was no specific error type for this. If the error is not the one, the following code should not be invoked and return early as the L72 does. 

Of course I could check like this `strings.Contains("unknown field")`, but it seemed ad-hoc. Any idea do you have to safely consolidate them?

Comment (hyeonLewis) [reply blockchain/gen_genesis.go:73]: I thought something like `strings.Contains`, but it seems unnecessary. Could you add a little comment about this?

Comment (hyunsooda) [reply blockchain/gen_genesis.go]: Refactored. PTAL.

**Review by hyeonLewis — APPROVED** [review]

**Comment by blukat29** [comment]
@hyunsooda  Can you move the check to `cmd/utils/nodecmd/chaincmd.go:initGenesis`? The `blockchain/gen_genesis.go` file is autogenerated, so it says DO NOT EDIT on top.

**Comment by hyunsooda** [comment]
Unfortunately, that was not available because `Genesis.UnmarhsalJSON()` is defined and overwritten the unmarshal behavior. Thus, `json.NewDecoder()` just gets called two times. There was a [suggestion](https://github.com/fjl/gencodec/issues/18) to add `DisallowUnknownFields`, but not been resolved yet.

**Comment by blukat29** [comment]
@hyunsooda I see. Thanks.

**Review by blukat29 — APPROVED** [review]

**Pull request merged**